v2/kafka-to-bigquery/src/test/java/com/google/cloud/teleport/v2/templates/KafkaToBigQueryFlexAvroIT.java

```diff
@@ -130,6 +144,19 @@ public void testKafkaToBigQueryAvroWithSchemaRegistry() throws IOException, Rest
                 .addParameter("outputDataset", bqDatasetId));
   }
 
+  @Test
+  public void testKafkaToBigQueryAvroWithSchemaRegistryWithKey() throws IOException, RestClientException {
+    baseKafkaToBigQueryAvro(
+        b ->
+            b.addParameter("avroFormat", "CONFLUENT_WIRE_FORMAT")
+                .addParameter(
+                    "schemaRegistryConnectionUrl",
+                    SecretManagerUtils.getSecret(
+                        "projects/269744978479/secrets/kafka-schema-registry-connection-url/versions/1"))
+                .addParameter("outputDataset", bqDatasetId)
+                .addParameter("useKafkaKey", "true"));
```

Comment:
How/where do we assert if key is getting added or not?

Reply:
Added assertion for checking a few rows.